tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -667,6 +677,9 @@ void EnqueueProgramCommand::assemble_runtime_args_commands() {
         // BRISC, NCRISC, TRISC...
         runtime_args_fetch_size_bytes += cmds.size_bytes();
     }
+    for (CoreType core_type : core_types) {
+        RecordKernelGroups(program, core_type, program.get_kernel_groups(core_type));
+    }
```

Comment:
Is there a reason we do this in the assemble_runtime_args function? This seems to be for kernel groups which doesn't directly map to rt args.

Reply:
Yeah moved it out of this function, thanks for catching